workflow/artifacts/git/git.go

```diff
@@ -58,9 +58,20 @@ func writePrivateKey(key string, insecureIgnoreHostKey bool) error {
 		return errors.InternalWrapError(err)
 	}
 	sshDir := fmt.Sprintf("%s/.ssh", usr.HomeDir)
-	err = os.Mkdir(sshDir, 0700)
-	if err != nil {
-		return errors.InternalWrapError(err)
+
+	if _, err = os.Stat(sshDir); err != nil {
+
+		if os.IsNotExist(err) {
+
+			err = os.Mkdir(sshDir, 0700)
+			if err != nil {
+				return errors.InternalWrapError(err)
+			}
+
+		} else {
+			return errors.InternalWrapError(err)
+		}
+
```

Comment:
We should just change `os.Mkdir` to `os.MkdirAll` which is an no-op if the directory already exists. The fix would then just be a one line change.